lib/array.js

```diff
@@ -21,18 +21,20 @@ module.exports = (api) => {
     let count = 0;
     done = api.metasync.cb(done);
 
-    items.forEach((item, index) => {
-      fn(item, (err, value) => {
-        if (errored) return;
-        if (err) {
-          errored = true;
-          return done(err);
-        }
-        result[index] = value;
-        count++;
-        if (count === len) done(null, result);
-      });
-    });
+    const callback = (index, err, value) => {
+      if (errored) return;
+      if (err) {
+        errored = true;
+        return done(err);
+      }
+      result[index] = value;
+      count++;
+      if (count === len) done(null, result);
+    };
+
+    for (let i = 0; i < len; i++) {
```

Comment:
@nechaido Declare loop variable outside of for `let i; for (i ...)` not a good syntax but we use it for performance everywhere.